src/Microsoft.TemplateEngine.Cli/HostSpecificDataLoader.cs

```diff
@@ -1,38 +1,47 @@
-using System.IO;
+using System;
+using System.Collections.Generic;
 using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Json;
 using Microsoft.TemplateEngine.Abstractions.Mount;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+using Microsoft.TemplateEngine.Utils;
 
 namespace Microsoft.TemplateEngine.Cli
 {
     public class HostSpecificDataLoader : IHostSpecificDataLoader
     {
-        public HostSpecificDataLoader(ISettingsLoader settingsLoader)
+        public HostSpecificDataLoader(ISettingsLoader settingsLoader, IJsonDocumentObjectModelFactory jsonDomFactory)
         {
             _settingsLoader = settingsLoader;
+            _jsonDomFactory = jsonDomFactory;
         }
 
-        private ISettingsLoader _settingsLoader;
+        private readonly ISettingsLoader _settingsLoader;
+        private readonly IJsonDocumentObjectModelFactory _jsonDomFactory;
 
         public HostSpecificTemplateData ReadHostSpecificTemplateData(ITemplateInfo templateInfo)
         {
             IMountPoint mountPoint = null;
 
             try
             {
+                HostSpecificTemplateData hostData = new HostSpecificTemplateData();
+
                 if (_settingsLoader.TryGetFileFromIdAndPath(templateInfo.HostConfigMountPointId, templateInfo.HostConfigPlace, out IFile file, out mountPoint))
                 {
-                    JObject jsonData;
-                    using (Stream stream = file.OpenRead())
-                    using (TextReader textReader = new StreamReader(stream, true))
-                    using (JsonReader jsonReader = new JsonTextReader(textReader))
+                    if (_jsonDomFactory.TryLoadFromFile(file, out IJsonToken root) && root is IJsonObject rootObject)
```

Comment:
Given where we want to go with this, maybe sticking with extensions that take stream or IPhysicalFileSystem and zip path and path to entry or at least physical file system + path

Reply:
The extension method name is bad - in this method, the variable "file" is actually an IFile, so I think we're ok as-is. I'll rename it to TryLoadFromIFile() 

I'll also get rid of the file-based version of the extension method.